scripts/init.sh

```diff
@@ -77,24 +77,26 @@ if [ -r $DEFAULT ]; then
     source $DEFAULT
 fi
 
-function pidofproc() {
-    if [ $# -ne 3 ]; then
-        echo "Expected three arguments, e.g. $0 -p pidfile daemon-name"
-    fi
+if ! typeset -F pidofproc &>/dev/null; then
+  function pidofproc() {
+      if [ $# -ne 3 ]; then
+          echo "Expected three arguments, e.g. $0 -p pidfile daemon-name"
+      fi
 
-    PID=`pgrep -f $3`
-    local PIDFILE=`cat $2`
+      pid=`pgrep -f '\'$3'\>'`
```

Comment:
Your example *should* also work, but I've had issues getting that to work in the past (thought I just tested locally and couldn't get it to work). To be honest I don't remember where I learned this syntax (TLDP, likely), I just know it works. As I'm sure you know, using regex with command line utilities can be... {unreliable, weird, frustrating}.